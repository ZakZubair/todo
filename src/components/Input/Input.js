import React from 'react';
import PropTypes from 'prop-types';

const Input = ({
  value, placeholder, onChange, onKeyPress, onClick, className,
}) => (
  <input
    className={className}
    value={value}
    onChange={onChange}
    placeholder={placeholder}
    onKeyPress={onKeyPress}
    onClick={onClick}
  />
);

Input.defaultProps = {
  placeholder: '',
  onKeyPress: null,
  onClick: null,
  className: 'input',
};

Input.propTypes = {
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onKeyPress: PropTypes.func,
  onClick: PropTypes.func,
  className: PropTypes.string,
};
export default Input;
