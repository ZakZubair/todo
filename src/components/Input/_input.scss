input {
  &.input {
    flex: 1;
    font-size: rem(18);
    color: $bright-gray;
    outline: none;
    border: none;
    padding-bottom: rem(16);
    margin: rem(24) 0 0;
    border-bottom: rem(2) solid $emerald-20;
    background-color: transparent;
    &::placeholder {
      color: $french-gray;
    }
  }
}
