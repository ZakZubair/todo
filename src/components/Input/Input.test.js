import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Input from './Input';

describe('test Input component', () => {
  it('test change event', () => {
    const mockCallBack = jest.fn();

    const wrapper = shallow((<Input onChange={mockCallBack} value="Old value" />));
    wrapper.find('input').simulate('change', { target: { value: 'New value' } });
  });

  it('render correctly input component', () => {
    const mockCallBack = jest.fn();

    const props = {
      value: 'Test value',
      onChange: mockCallBack,
      onKeyPress: mockCallBack,
      onClick: mockCallBack,
    };
    const InputComponent = renderer.create(<Input {...props} />).toJSON();
    expect(InputComponent).toMatchSnapshot();
  });
});
