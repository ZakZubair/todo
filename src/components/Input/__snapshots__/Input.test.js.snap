// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Input component render correctly input component 1`] = `
<input
  className="input"
  onChange={[MockFunction]}
  onClick={[MockFunction]}
  onKeyPress={[MockFunction]}
  placeholder=""
  value="Test value"
/>
`;
